src/test/java/me/karakelley/http/handlers/StaticFilesHandlerTest.java

```diff
@@ -1,59 +1,152 @@
 
-      ContentPresenter contentPresenter = new HtmlPresenter();
-      Handler handler = new StaticFilesHandler(publicDirectory, contentPresenter);
-      Response response = handler.respond(new Request(HttpMethod.GET, "/", "HTTP/1.1", null, null,0));
 
-      assertTrue(new String(response.getBody()).split("", 2).length == 2);
-    });
+  @Nested
+  class WhenRequestIsGet {
+    @Test
+    void testDisplaysFiles() {
+      TempFilesHelper.withTempDirectory(directory -> {
+        Path fileOne = TempFilesHelper.createTempFile(directory, "/test1");
+        Path fileTwo = TempFilesHelper.createTempFile(directory, "/test2");
+        PublicDirectory publicDirectory = PublicDirectory.create(directory.toString());
+        Handler handler = new StaticFilesHandler(publicDirectory, new HtmlPresenter());
+        Response response = handler.respond(new Request(HttpMethod.GET, "/", "HTTP/1.1", null, null,0));
+
+        assertTrue(new String(response.getBody()).split("", 2).length == 2);
+      });
+    }
+
+    @Test
+    void testCreatesLinksForDirectories() {
+      TempFilesHelper.withTempDirectory(directory -> {
+        Path file = TempFilesHelper.createTempFile(directory, "/test1");
+        PublicDirectory publicDirectory = PublicDirectory.create(directory.toString());
+        Handler handler = new StaticFilesHandler(publicDirectory, new HtmlPresenter());
+        Response response = handler.respond(new Request(HttpMethod.GET, "/", "HTTP/1.1", null, null, 0));
+
+        assertTrue(new String(response.getBody()).contains("<p><a href=\"/test1.txt\">test1.txt</a></p>"));
+      });
+    }
+
+    @Test
+    void testServesTextFile() {
+      TempFilesHelper.withTempDirectory(directory -> {
+        Path file = TempFilesHelper.createTempFile(directory, "/test1");
+        TempFilesHelper.createContents("Hello World", file);
+        PublicDirectory publicDirectory = PublicDirectory.create(directory.toString());
+        Handler handler = new StaticFilesHandler(publicDirectory, new HtmlPresenter());
+        Response response = handler.respond(new Request(HttpMethod.GET, "/test1.txt",  "HTTP/1.1", null, null, 0));
+
+        assertEquals("Hello World", new String(response.getBody()));
+      });
+    }
   }
 
-  @Test
-  void testCreatesLinksForDirectories() {
-    TempFilesHelper.withTempDirectory(directory -> {
-      Path file = TempFilesHelper.createTempFile(directory, "/test1");
-      PublicDirectory publicDirectory = PublicDirectory.create(directory.toString());
-      ContentPresenter contentPresenter = new HtmlPresenter();
-      Handler handler = new StaticFilesHandler(publicDirectory, contentPresenter);
-      Response response = handler.respond(new Request(HttpMethod.GET, "/", "HTTP/1.1", null, null, 0));
-
-      assertTrue(new String(response.getBody()).contains("<p><a href=\"/test1.txt\">test1.txt</a></p>"));
-    });
+  @Nested
+  class WhenRequestIsPost {
+    @Test
+    void test201ResponseForPostContainsLocationHeader() {
```

Comment:
Is that what this is even testing?

Reply:
Not quite - I will update this to be more expressive